src/shell.rs

```diff
@@ -22,6 +28,43 @@ impl<'a> Shell<'a> {
     pub fn path_for(&self, filename: &str) -> PathBuf {
         self.root_path().join(filename)
     }
+
+    pub fn create_links<P: AsRef<Path>>(&self, base_directory: P) -> FileOperations {
+        let mut file_ops = FileOperations::rooted_at(base_directory);
+        let entries = fs::read_dir(self.root_path()).unwrap();
+        let shell_path_length = self.root_path().to_str().unwrap().len();
+        let file_root_length = file_ops.root.to_str().unwrap().len();
+        for entry in entries {
+            let entry = entry.unwrap().path();
+            let entry = entry.to_str().unwrap();
+            let rel_shell = &entry[shell_path_length+1..entry.len()];
+            let rel_file_root = &entry[file_root_length+1..entry.len()];
+            file_ops.link(rel_file_root, rel_shell);
+        }
+        return file_ops;
+    }
+
+    pub fn remove_links<P: AsRef<Path>>(&self, base_directory: P) -> FileOperations {
+        let mut file_ops = FileOperations::rooted_at(&base_directory);
+        for entry in walkdir::WalkDir::new(&base_directory) {
+            let entry = entry.unwrap();
+            match fs::read_link(entry.path()) {
+                Ok(i) => {
+                    let fname = entry.path().file_name().unwrap().to_str().unwrap();
+                    let link_path = base_directory.as_ref().join(&i);
+                    let shell_path = self.path_for(fname);
+                    if link_path == shell_path {
+                        let entry_str = entry.path().to_str().unwrap();
+                        let base_str = base_directory.as_ref().to_str().unwrap();
```

Comment:
So writing this with `unwrap` everywhere is a great way to prototype it, but for production code we want to get rid of almost all of these.  Since this is a utility, we probably want to have it return a `Result<FileOperations>` and use the `try!` macro to exit early on errors. An excellent post about good error handling in Rust http://blog.burntsushi.net/rust-error-handling/